internal/internal_workflow_testsuite.go

```diff
@@ -2686,8 +2701,9 @@ func (env *testWorkflowEnvironmentImpl) resolveNexusOperation(seq int64, token s
 			panic(fmt.Errorf("no running operation found for sequence: %d", seq))
 		}
 		if err != nil {
+			err = convertKnownErrors(err)
 			failure := env.failureConverter.ErrorToFailure(err)
-			err = env.failureConverter.FailureToError(nexusOperationFailure(handle.params, handle.operationToken, failure.GetCause()))
+			err = env.failureConverter.FailureToError(nexusOperationFailure(handle.params, handle.operationToken, failure))
```

Comment:
@bergundy I think this was missed in https://github.com/temporalio/sdk-go/pull/1822

Reply:
I don't think we need to convert known errors here, this is just used when a handler workflow completes and it should already include a properly structured failure error.
I think the reason I use `GetCause()` here is that the workflow error was wrapping the failure and I didn't want that to be part of the cause chain to mimic the server behavior.
Arguably unwrapping should be done where this function is called though.
Worth double checking this.

Reply:
This is also called when the workflow failed to start.
When executing a child workflow, there's a `callback` and a `startedHandler`, and they are both called in case the child workflow fails to start.

Reply:
So the unwrapping should probably be done at the call site.